### PR TITLE
Base image for rails3 / frontend

### PR DESCRIPTION
**Pull request by raganw**

### CODE REVIEW

**Comment by rantler** [comment]
I'm running into the same build issue with snappy with this.

**Comment by rantler** [comment]
I had to do the following steps within the docker container in order to get it working for me in boot2docker:
```
apt-get update
apt-get install automake
apt-get install libxml2-dev
apt-get install libxslt1-dev
apt-get install zlib1g-dev
apt-get install gcc
apt-get install g++
apt-get install patch
apt-get install libsnappy1
apt-get install libsnappy-dev

gem install nokogiri -- --use-system-libraries=true --with-xml2-include=/usr/include/libxml2
gem install multi_xml -- --use-system-libraries=true --with-xml2-include=/usr/include/libxml2
gem install nokogumbo -- --use-system-libraries=true --with-xml2-include=/usr/include/libxml2
gem install snappy
```

**Comment by rantler** [comment]
:shipit:!

**Comment by raganw** [comment]
@rantler This has been updated to include inserting the `Gemfile` and `Gemfile.lock` for the frontend, and running `bundle install`, to prime the gem cache.  This will greatly speed up packaging the frontend codebase into this image.

File: ruby/rails3-deps/Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,269 @@
+GIT
+  remote: https://github.com/socrata-platform/statsd-ruby-fork.git
```
Comment (rantler) [inline ruby/rails3-deps/Gemfile.lock]: :+1:

File: ruby/rails3-deps/Gemfile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+gem 'rack', '~>1.4.5'
+gem 'rails', '~>3.2.12', :require => nil
+gem 'recaptcha', :require => 'recaptcha/rails'
+gem 'redcarpet'
+gem 'rinku', :require => 'rails_rinku'
+gem 'sass', '~>3.2'
+gem 'stomp', '1.1.6'
+gem 'sanitize'
+gem 'snappy'
+gem 'statsd-ruby', :require => 'statsd', :git => 'https://github.com/socrata-platform/statsd-ruby-fork.git'
```
Comment (rantler) [inline ruby/rails3-deps/Gemfile]: :+1:

**Comment by rantler** [comment]
Looks good to me. My boot2docker instance isn't behaving well, so I wasn't able to build the Docker image. Nevertheless, I say, :shipit:!

**Pull request merged**